Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs

```diff
@@ -514,7 +527,38 @@ public static IEnumerable<TestCase> ConnectionStringTestCases()
                     { SFSessionProperty.DISABLEQUERYCONTEXTCACHE, defDisableQueryContextCache },
                     { SFSessionProperty.DISABLE_CONSOLE_LOGIN, defDisableConsoleLogin },
                     { SFSessionProperty.ALLOWUNDERSCORESINHOST, "false" },
-                    { SFSessionProperty.QUERY_TAG, testQueryTag }
+                    { SFSessionProperty.QUERY_TAG, testQueryTag },
+                    { SFSessionProperty.ALLOW_SSO_TOKEN_CACHING, defAllowSSOTokenCaching }
+                }
+            };
+            var testCaseWithAllowSSOTokenCaching = new TestCase()
```

Comment:
let's not continue this way of testing, because it has a lot of disadvantages. For example if you update your PR with the changes from master you will have plenty of conflicts in this file - that is because many parameters which do not related to each other are tested together.
Instead you could create a separate test for testing ALLOW_SSO_TOKEN_CACHING property similar to TestValidateCorrectAccountNames or TestThatItFailsForWrongConnectionParameter. In such test we get as input a connection string or its part and assert only on the property which we want to verify.

Reply:
That makes sense,, a separate test is now created for the new property